ext/tracer_tag_propagation/tracer_tag_propagation.c

```diff
@@ -79,15 +79,46 @@ void ddtrace_add_tracer_tags_from_array(zend_array *array, zend_array *root_meta
     ZEND_HASH_FOREACH_END();
 }
 
-void ddtrace_get_propagated_tags(zend_array *tags) {
+static zend_array *ddtrace_get_propagated() {
     zend_array *propagated = &DDTRACE_G(propagated_root_span_tags);
+    ddtrace_root_span_data *root_span = DDTRACE_G(active_stack)->root_span;
+    if (root_span) {
+        propagated = ddtrace_property_array(&root_span->property_propagated_tags);
+    }
+
+    return propagated;
+}
+
+static zend_array *ddtrace_get_root_meta() {
     zend_array *root_meta = &DDTRACE_G(root_span_tags_preset);
     ddtrace_root_span_data *root_span = DDTRACE_G(active_stack)->root_span;
     if (root_span) {
         root_meta = ddtrace_property_array(&root_span->property_meta);
-        propagated = ddtrace_property_array(&root_span->property_propagated_tags);
     }
 
+    return root_meta;
+}
+
+zval *ddtrace_propagated_tags_get_tag(const char *tag) {
+    zend_array *propagated = ddtrace_get_propagated();
+    zend_array *root_meta = ddtrace_get_root_meta();
+
+    if (!tag) {
```

Comment:
nit: could be done before the calls to ddtrace_get_propagated / ddtrace_get_root_meta to save a few CPU instructions

Reply:
Done